numpy/random/generator.pyx

```diff
@@ -3827,6 +3828,7 @@ cdef class Generator:
                     buf[...] = x[j]
                     x[j] = x[i]
                     x[i] = buf
+            x = np.swapaxes(x, 0, axis)
         else:
             # Untyped path.
```

Comment:
This needs some thought.  Perpahs you should raise TypeError here since it isn't really possible to use axis with things that aren't ndarray or subclasses.

Reply:
Maybe there should also be a `numpy.AxisError` in the "fast path" above if `axis` is neither 0 nor -1.

Reply:
@bashtage Do you mean raising `TypeError` for `# Untyped path`? Because `x` has already been confirmed as a `ndarray` in the other two cases.

Reply:
@eric-wieser In this case, we can make sure it is an `ndarray`, do we still need to check `if axis != 0`? I am just wondering when will it fail.

Reply:
@kianasun: I am talking about the `# Untyped path` which this comment chain is in response to.

Reply:
@eric-wieser I see. I will add this for `Untyped path`

Reply:
Should there be a test case with axis != 0 that hits the untyped path?

Reply:
@wkschwartz I added one for `random.shuffle`.